### ISSUE TITLE
Add pypy3 support or fix compatibility with python 3.6

### ISSUE THREAD

**Issue by ppolewicz** [pyca/cryptography#2880]
I have found this: https://github.com/pyca/cryptography/pull/2012
but as far as I see here: http://cffi.readthedocs.io/en/latest/using.html#python-3-support cffi seems to support pypy3 already.

We'd like to use cryptography in b2 CLI (https://github.com/Backblaze/B2_Command_Line_Tool/), but we support all kinds of pythons and it was found that:
- Cryptography 0.9.3 supports pypy2 and pypy3, but not python 3.6 (https://travis-ci.org/Backblaze/B2_Command_Line_Tool/builds/126348171)
- Cryptography 1.3.1 supports pypy2 and python 3.6, but not pypy3 (https://travis-ci.org/Backblaze/B2_Command_Line_Tool/builds/126345892)

The latter problem seems to be caused by https://github.com/pyca/cryptography/blob/master/setup.py#L50:
```python
if platform.python_implementation() == "PyPy":
    if sys.pypy_version_info < (2, 6):
        raise RuntimeError(
            "cryptography 1.0 is not compatible with PyPy < 2.6. Please "
            "upgrade PyPy to use this library."
        )
```
which doesn't account for the fact that PyPy2 version is already 5.x, while PyPy3 version number is still at 2.4 (http://doc.pypy.org/en/latest/release-pypy3-2.4.0.html).

Would you please consider supporting PyPy3 on current head branch or python 3.6 (nightly) in 0.9.4?

**Comment by reaperhulk** [comment]
PyPy integrates cffi directly into its distribution, so the version of cffi that's available is fixed at release (and can't be upgraded via pip). Unfortunately, PyPy3's cffi is `0.8.6` and the out-of-line API compilation we use as of cryptography 1.0 was introduced with cffi 1.0+. This means we can't support PyPy3 in master again until a new release occurs.

We're unlikely to issue a patch to 0.9.3 to support an unreleased Python, but I'll look at what the change is.

One potential workaround on your end is to use `platform.python_implementation()` and `platform.python_version()` to detect PyPy3 and set your cryptography version requirement appropriately. An unpleasant hack, but PyPy3 is the red-headed stepchild of PyPy right now :(

**Comment by ppolewicz** [comment]
Thank you very much for the information.

Issue on python 3.6 is that `cryptography/src/cryptography/utils.py`, line 42, in verify_interface, does this:
```python
      spec = inspect.getargspec(getattr(iface, method))
```
while `inspect.getargspec()` is [deprecated since python 3.0](https://docs.python.org/3.6/library/inspect.html#inspect.getargspec). I tried to fix it, but apparently 44878a53 already resolved this issue.

Is there a release scheduled?

**Comment by alex** [comment]
That commit was released in cryptography 1.1, which came out serveral months ago

**Comment by ppolewicz** [comment]
I see. Thank you.
